### PR TITLE
add POA verification to Appeals API

### PR DESCRIPTION
**Pull request by charleystran**
## Description of change
This pull request is to add power of attorney validation to the Appeals API similar to how we validate the Power of attorney in the Claims API

## Testing done
Testing was done via the test suit as well as testing with the mock responses from EVSS that were generated for the Claims POA validation

## Acceptance Criteria (Definition of Done)
A proper power of attorney should be able to retrieve Appeals Data
A non matching Power of Attorney should be rejected as unauthorized

#### Unique to this PR

- [x] A proper power of attorney should be able to retrieve Appeals Data
- [x] A non matching Power of Attorney should be rejected as unauthorized

#### Applies to all PRs

- [x] Appropriate logging
- [x] Swagger docs have been updated, if applicable
- [x] https://github.com/department-of-veterans-affairs/vets-contrib/issues/892
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)
- [x] Provide which alerts would indicate a problem with this functionality (if applicable)

### CODE REVIEW

File: lib/evss/power_of_attorney_verifier.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module EVSS
+  class PowerOfAttorneyVerifier
+    def initialize(user)
+      @user = user
+      @use_mock = Settings.evss.power_of_attorney_verifier || false
```
Comment (ATeal) [inline lib/evss/power_of_attorney_verifier.rb]: I *think* in most cases the settings specify whether to mock it or not (e.g. like `mock_vso_search`)

Also don't forget to make a devops pr for mocking in dev

File: lib/evss/power_of_attorney_verifier.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      @veteran = Veteran.new(@user)
+    end
+
+    def verify(header)
```
Comment (ATeal) [inline lib/evss/power_of_attorney_verifier.rb]: I would call this consumer_id or something like that rather than header is a little more ambiguous

Comment (charleystran) [reply lib/evss/power_of_attorney_verifier.rb]: Changes have been made

File: modules/appeals_api/app/controllers/appeals_api/v0/appeals_controller.rb [hunk 3]
```diff
@@ -66,7 +66,7 @@ def header(key)
 
       def va_profile
         OpenStruct.new(
-          birth_date: header('X-VA-Birth-Date')
+          birth_date: (header('X-VA-Birth-Date') || nil)
```
Comment (ATeal) [inline modules/appeals_api/app/controllers/appeals_api/v0/appeals_controller.rb]: Will EVSS work with a nil birth date? If so why are we requiring it in claims lol

Comment (bastosmichael) [reply modules/appeals_api/app/controllers/appeals_api/v0/appeals_controller.rb]: we don't need to fall back to nil I agree here, is there a reason this was done? were you getting a blank string value by mistake?

**Comment by bastosmichael** [comment]
So @charleystran Make the header 'X-Consumer-PoA' for now and also do it so that the non SSN fields required only fail when the poa header isn't around and document accordingly. This will allow folks like the National Call Center and others to still only make appeals requests using just SSN as VIP users while requiring VSO's to include everything.

File: modules/appeals_api/app/controllers/appeals_api/v0/appeals_controller.rb (hunk 3 above)
Comment (charleystran) [reply modules/appeals_api/app/controllers/appeals_api/v0/appeals_controller.rb]: I removed the whole va_profile as it is not needed from the appeals side

**Pull request merged**